modules/apps/image/image-service/src/main/java/com/liferay/image/service/internal/upgrade/v1_0_0/ImageStorageUpgradeProcess.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.image.service.internal.upgrade.v1_0_0;
+
+import com.liferay.document.library.kernel.store.DLStoreUtil;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.orm.ActionableDynamicQuery;
+import com.liferay.portal.kernel.dao.orm.RestrictionsFactoryUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Image;
+import com.liferay.portal.kernel.service.ImageLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.io.InputStream;
+
+/**
+ * @author Adolfo Pérez
+ */
+public class ImageStorageUpgradeProcess extends UpgradeProcess {
+
+	public ImageStorageUpgradeProcess(ImageLocalService imageLocalService) {
+		_imageLocalService = imageLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws PortalException {
+		ActionableDynamicQuery actionableDynamicQuery =
+			_imageLocalService.getActionableDynamicQuery();
+
+		actionableDynamicQuery.setAddCriteriaMethod(
+			dynamicQuery -> dynamicQuery.add(
+				RestrictionsFactoryUtil.eq("companyId", 0L)));
+		actionableDynamicQuery.setPerformActionMethod(
+			(Image image) -> {
+				String fileName = _getFileName(image);
```

Comment:
I looks great to me.

The only thing I can add is that maybe we can skip the try  + add/delete id (image.getCompanyId() == _DEFAULT_COMPANY_ID) but I do not know if there will be a lot of cases or all will be different

thanks for all!

Reply:
Not sure about your question. Do you suggest to get rid of the try/catch block, and let if fail if something goes wrong?

The try/catch is there to avoid 1 failure aborting the whole upgrade process. Sometimes, files may be missing in the filesystem, or maybe the image in the db is old and crusty. With that in place, the error is logged, but we continue, doing as much work as possible.